rdsbroker/broker.go

```diff
@@ -391,10 +391,57 @@ func (b *RDSBroker) LastOperation(instanceID string) (brokerapi.LastOperationRes
 	return lastOperationResponse, nil
 }
 
+func (b *RDSBroker) CheckAndRotateCredentials() error {
+	dbInstanceDetailsList, err := b.dbInstance.DescribeByTag("Broker Name", b.brokerName)
+	if err != nil {
+		b.logger.Error("Could not retrieve the list of DB instances", err)
+		return err
+	}
+
+	for _, dbInstanceDetails := range dbInstanceDetailsList {
+		dbAddress := dbInstanceDetails.Address
+		dbPort := dbInstanceDetails.Port
+		masterUsername := dbInstanceDetails.MasterUsername
+		serviceInstanceID := b.dbInstanceIdentifierToServiceInstanceID(dbInstanceDetails.Identifier)
+
+		var dbName string
+		if dbInstanceDetails.DBName != "" {
+			dbName = dbInstanceDetails.DBName
```

Comment:
The [rest of the code](https://github.com/alphagov/paas-rds-broker/blob/73d2b88af73894de0767fe36192716d836034ffb/rdsbroker/broker.go#L235-L239) always assume this can be empty. I think it's safer to leave it like this as it may not be reliable. An RDS instance may not hold any databases for example.